### PR TITLE
Chef Produce Console, for cases where there are no botanists

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

For some god forsaken reason, icons don't work right now. Here's what it looks like with no icons!
image

The chef now has a console that they can load up groceries. The next time the cargo shuttle lands at the station it will have a crate with everything the chef ordered. The chef can also pay for a 25% upcharge on their total cart for an instant delivery, though this is pretty damn expensive so is really not too worth it outside of missing one ingredient in a recipe that you wanna quickly nab.

#### Botanists are still the BEST way to get good produce!

one side of this done, the actual ordering needs to be finished.

## Why It's Good For The Game

Requested by @Qustinnus , chef runs out of food for this job if there are no botanists.

## Changelog
:cl:
add: The Chef now has a produce console for ordering ingredients! Botanists are still the best way to get stuff, though.
/:cl:

### CODE REVIEW

File: _maps/map_files/KiloStation/KiloStation.dmm [hunk 1]
```diff
@@ -65525,6 +65525,10 @@
 /obj/item/food/dough,
 /obj/item/kitchen/rollingpin,
 /obj/item/kitchen/rollingpin,
+/obj/item/paper/crumpled{
+	info = "Hey whoever designed this shithole didn't give us space to install the produce computer so it's in maintenance near the theatre.";
```
Comment (Qustinnus) [inline _maps/map_files/KiloStation/KiloStation.dmm:65529]: :trollface:

File: code/game/machinery/computer/chef_orders/chef_order.dm [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+			"cost" = item.cost_per_order,
+			"amt" = grocery_list[item]
+			))
+
+/obj/machinery/computer/chef_order/ui_act(action, params)
+	. = ..()
+	if(.)
+		return
+	var/mob/living/chef = usr
+	if(!istype(chef))
```
Comment (Qustinnus) [inline code/game/machinery/computer/chef_orders/chef_order.dm:58]: typecheck then cast you crazy man

Comment (tralezab) [reply code/game/machinery/computer/chef_orders/chef_order.dm:58]: It is technically the same outcome, just different syntax

File: _maps/map_files/KiloStation/KiloStation.dmm (hunk 1 above)
Comment (tralezab) [reply _maps/map_files/KiloStation/KiloStation.dmm:65529]: :trollface:

**Comment by ArcaneMusic** [comment]
Opinion: might be better and faster space-wise to shove this into a modular computer app and give the kitchen a laptop or something with it installed instead.

Other than that, seems neat.

File: code/game/machinery/computer/chef_orders/chef_order.dm (hunk 2 above)
Comment (LemonInTheDark) [reply code/game/machinery/computer/chef_orders/chef_order.dm:58]: one is worse

**Comment by bluetshirtguy** [comment]
Why can't the chef just use the public gardens or build/steal a hydroponics tray? Getting rid of the middle man for lazy chefs isn't a good thing.

**Comment by tralezab** [comment]
Cutting out the middleman for the chef is not about playing solo, it's about still allowing departmental cooperation. A chef shouldn't be forced to play two jobs at once just to be able to do the other!

**Comment by Krysonism** [comment]
QoL creeping away half of chef gameplay.

**Comment by Krysonism** [comment]
> Cutting out the middleman for the chef is not about playing solo, it's about still allowing departmental cooperation. A chef shouldn't be forced to play two jobs at once just to be able to do the other!

Sourcing ingredients is chef gameplay.

**Comment by tralezab** [comment]
>No botanists
>Run out of food 10 minutes in
>Now you're growing 20 potency plants in public garden halfway across the station
>income goes down by 90% because nobody joined your job

And this is worse than cargo co-departmental cooperation when usually botanists just give you everything you need?

**Comment by MrDoomBringer** [comment]
the cargo cooperation here is based, it makes sense to have the option between cooperating with botany or cargo

**Comment by tralezab** [comment]
More like a backup plan, ofc you get your products for free working with botany ignoring the cargo wait times

**Comment by maxymax13** [comment]
finally, lowpop chef can now make something other than the things in the fridges roundstart

File: _maps/map_files/KiloStation/KiloStation.dmm (hunk 1 above)
Comment (interestingusernam3) [reply _maps/map_files/KiloStation/KiloStation.dmm:65529]: :trollface:

**Comment by cacogen** [comment]
I was hoping to see something like this added. You'll always be using the express option though because cargo is almost never reliable.

File: _maps/map_files/KiloStation/KiloStation.dmm (hunk 1 above)
Comment (cacogen) [reply _maps/map_files/KiloStation/KiloStation.dmm:65529]: How are you doing that with your mouth

**Comment by tralezab** [comment]
We'll see, it's quite the upcharge, haha

**Comment by cacogen** [comment]
Given the time limit on orders if you need a couple of tomatoes for pasta and there's no one to grow them you're either using the express option, spending ten minutes growing them yourself or forcing your way into cargo or the cargo shuttle to get your crate.

File: tgui/packages/tgui/interfaces/Cargo.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import { toArray } from 'common/collections';
 import { useBackend, useSharedState } from '../backend';
-import { AnimatedNumber, Box, Button, Flex, LabeledList, Section, Table, Tabs } from '../components';
+import { AnimatedNumber, Box, Button, Divider, Flex, LabeledList, NoticeBox, Section, Table, Tabs } from '../components';
```
Comment (stylemistake) [inline tgui/packages/tgui/interfaces/Cargo.js:3]: Why did you add these imports?

**Review by stylemistake — APPROVED** [review]

**Comment by Time-Green** [comment]
Honestly literally every job needs this, it makes it really easy to place job specific orders. This is the future of cargo-station integration. 

I'm not asking you to do out-of-scope stuff, just politely asking you to keep the code sexy and flexible so that someone else can (me if I dont forget).

File: tgui/packages/tgui/interfaces/Cargo.js (hunk 3 above)
Comment (tralezab) [reply tgui/packages/tgui/interfaces/Cargo.js:3]: oops!

**Comment by Tlaltecuhtli** [comment]
make it send cargo dep a message  if chef made an order

**Comment by tralezab** [comment]
@Tlaltecuhtli aight sure

File: code/game/machinery/computer/chef_orders/chef_order.dm [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+/obj/machinery/computer/chef_order
+	name = "Produce Orders Console"
+	desc = "An interface for ordering fresh produce and other. A far more expensive option than the botanists, but oh well."
+	icon_screen = "request"
+	icon_keyboard = "generic_key"
+	circuit = /obj/item/circuitboard/computer/chef_order
+	light_color = LIGHT_COLOR_ORANGE
+
+	COOLDOWN_DECLARE(order_cooldown)
+	var/list/order_datums = list()
```
Comment (Mothblocks) [inline code/game/machinery/computer/chef_orders/chef_order.dm:11]: Are you sure this should not be a static list? Is it different across computers?

File: code/game/machinery/computer/chef_orders/chef_order.dm [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+				return
+			say("Thank you for your purchase! Please note: The charge of this purchase and machine cooldown has been doubled!")
+			COOLDOWN_START(src, order_cooldown, 120 SECONDS)
+			var/obj/structure/closet/supplypod/bluespacepod/pod = new()
+			pod.explosionSize = list(0,0,0,0)
+			for(var/datum/orderable_item/item as anything in grocery_list)//every order
+				for(var/amt in 1 to grocery_list[item])//every order amount
+					new item.item_instance.type(pod)
+			var/turf/T = get_turf(chef)
+			new /obj/effect/pod_landingzone(T, pod)
```
Comment (Mothblocks) [inline code/game/machinery/computer/chef_orders/chef_order.dm:123]:
```suggestion
			var/turf/landing_location = get_turf(chef)
			new /obj/effect/pod_landingzone(landing_location, pod)
```

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Comment by cacogen** [comment]
I don't know if it changed since I last played it but it was missing spider eggs and soybeans and berries and other stuff I don't remember. 

Thankfully the express option is cheap enough that you don't have to rely on cargo, because that is a dumb idea in the context of the tourist mechanic.

**Comment by tralezab** [comment]
Luckily for you those were in (thanks for live testing to help me find out what I need)
